src/test/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMergeTest.java

```diff
@@ -282,7 +282,7 @@ public void testTriggerMerge() throws Exception {
         assertThat(merger.perform(build, null, listener)).isEqualTo(true);
 
         setupConditions(adminLogin, committerName, committerEmail, nonTriggerPhrase);
-        assertThat(merger.perform(build, null, listener)).isEqualTo(false);
+        assertThat(merger.perform(build, null, listener)).isEqualTo(true);
```

Comment:
I can encapsulate the "is there intention to merge?" into a separate method test that and then compare with "did merge?". The problem with checking `merger.perform()` return value is that it API documents it along the lines of "false if should not proceed to other publishers".
Could also check for build status changes (i.e. FAILURE) to make sure the build fails when it should, regardless of `perform` return value.